doc/doc/advanced/pgxl.md

```diff
@@ -0,0 +1,310 @@
+sudo apt-get -y install zlib1g-dev libreadline6-dev python-dev libssl-dev
+sudo localedef -i en_US -c -f UTF-8 -A /usr/share/locale/locale.alias en_US.UTF-8
+```
+
+Now set `$TARGET_DIR` to the directory into which you would like to install XL.
+```bash
+TARGET_DIR=/opt/pgxl
+```
+
+Then add the following line(s) to `/etc/ld.so.conf` (replace `$TARGET_DIR` with dir):
+```bash
+/usr/local/lib
+$TARGET_DIR/lib
+```
+
+The remaining parts of the installation do not require sudo rights. However, make sure
+the user account doing the installation has write permissions to `$TARGET_DIR`.
+
+Create a build directory and download the XL sources.
+```bash
+BUILD_DIR=~/pgxl_install
+mkdir -p $BUILD_DIR
+cd $BUILD_DIR
+wget -O pgxl-v9.2.tar.gz http://sourceforge.net/projects/postgres-xl/files/Releases/Version_9.2rc/postgres-xl-v9.2-src.tar.gz/download
+tar -xzf pgxl-v9.2.tar.gz
+```
+
+Now, you can build the sources.
+```bash
+cd $BUILD_DIR/postgres-xl/
+./configure --with-python --with-openssl --prefix $TARGET_DIR
+make -j 4
+make install
+```
+
+Additionally modules can be added as follows.
+```bash
+for pkg in btree_gin btree_gist earthdistance fuzzystrmatch hstore intagg intarray oid2name \
+pg_buffercache pgcrypto pgxc_clean pgxc_ctl pgxc_ddl pgxc_monitor stormstats \
+tablefunc tsearch2 unaccent; do
+    cd $BUILD_DIR/postgres-xl/contrib/$pkg
+    make; make install
+done
+```
+
+### Set XL related session variables
+
+To set Greenplum related session variables, modify your
+`~/.bashrc` script and add the line:
+
+```bash
+export PATH=$TARGET_DIR/bin:\$PATH
+export PGXC_CTL_HOME=$TARGET_DIR/conf
+```
+
+### Configure ssh with localhost
+
+Now you need to generate ssh keys for `localhost`. Run: 
+```bash
+mkdir -p ~/.ssh
+chmod 700 ~/.ssh
+cd ~/.ssh
+if [ ! -f id_rsa.pub ]; then
+    ssh-keygen -t rsa -N "" -f id_rsa
+fi
+cat id_rsa.pub >> authorized_keys
+chmod 600 authorized_keys
+```
+
+Then you should be able to `ssh` into `localhost` without password, and you can
+move on.
+
+### Create directories for the database
+
+Now set `$DATA_DIR` to the directory, where the database files will be stored.
+**Be sure that you have write permission to this directory**.
+
+```bash
+mkdir -p $DATA_DIR
+```
+
+### Configure the XL database
+
+In order to run XL, we still need to create a file with configuration settings. Run
+```bash
+mkdir -p $TARGET_DIR/conf
+```
+
+Now, assuming that `$USER` is set to the account under which pgxl should execute, the
+following settings should provide a good starting point.
+```bash
+USER_DB_PORT=5432
+MAX_USER_CONNECTIONS=100
+DATA_NODE_SHARED_BUFFERS="2000MB"
+DATA_NODE_WORK_MEM="128MB"
+DATA_NODE_MAINTENANCE_MEM="128MB"
+DATA_NODE_WAL_BUFFERS="16MB"
+DATA_NODE_CHECKPOINT_SEGMENTS="256"
+```
+
+Create a file `$TARGET_DIR/conf/pgxc_ctl.conf` with the following contents (replace
+variables with their values):
+
+```
+pgxcOwner=$USER
+pgxcUser=\$pgxcOwner
```

Comment:
There seems many unnecessary backslashes here for pgxc_ctl.conf.  Is this intentional?

I'd argue most steps involving config file creation should be written in the following style:

1. first ask the user to set several shell variables, 
2. then run a huge `cat` command with heredoc that produces the file.

It's easy for the reader since they can blindly copy/paste the whole thing into their terminal to just get things work and also clear how advanced readers could customize if they wanted.

For example:
```bash
# some customizable variables
USER=mylogin
pgxcOwner=mylogin
# more variables ...

# create pgxc_ctl.conf
cat >"$TARGET_DIR"/conf/pgxc_ctl.conf <<EOF
pgxcOwner=$USER
pgxcUser=$pgxcOwner
...
EOF
```